pkg/harvester-manager/list/harvesterhci.io.management.cluster.vue

```diff
@@ -111,19 +185,111 @@ export default {
+
+          return isHarvesterCluster(cluster);
+        })
+        .map((row) => {
+          if (row.isReady) {
+            row.setSupportedHarvesterVersion();
+          }
+
+          return row;
+        });
     },
 
     typeDisplay() {
-      return this.t(`typeLabel."${ HCI.CLUSTER }"`, { count: this.row?.length || 0 });
+      return this.t(`typeLabel."${ HCI.CLUSTER }"`, { count: this.rows?.length || 0 });
     },
   },
 
   methods: {
+    async getHarvesterRepository() {
+      try {
+        return await getHelmRepository(this.$store, HARVESTER_REPO.spec.gitRepo, HARVESTER_REPO.spec.gitBranch);
+      } catch (error) {
+        this.harvesterRepositoryError = true;
+      }
+    },
+
+    async setHarvesterLatestVersion(repository) {
+      try {
+        const chart = await getHelmChart(this.$store, repository, HARVESTER_CHART.name);
+
+        if (semver.gt(chart.version, this.harvester.extension.version)) {
+          this.harvesterLatestVersion = chart.version;
+        }
+      } catch (error) {
+        this.harvesterExtensionUpdateError = true;
+      }
+    },
+
+    async installHarvesterExtension(btnCb) {
+      let installed = false;
+
+      try {
+        const harvesterRepo = await ensureHelmRepository(this.$store, HARVESTER_REPO.spec.gitRepo, HARVESTER_REPO.metadata.name, HARVESTER_REPO.spec.gitBranch);
+
+        /**
+         * Server issue
+         * It needs to refresh the HelmRepository because the server can have a previous one in the cache.
+         */
+        await refreshHelmRepository(harvesterRepo);
+
+        const chart = await getHelmChart(this.$store, harvesterRepo, HARVESTER_CHART.name);
+
+        await installHelmChart(harvesterRepo, { ...HARVESTER_CHART, version: chart.version }, {}, UI_PLUGIN_NAMESPACE, 'install');
+
+        const extension = await waitForUIExtension(this.$store, HARVESTER_CHART.name);
+
+        installed = await waitForUIPackage(this.$store, extension);
+      } catch (error) {
+      }
+
+      this.harvesterExtensionInstallError = !installed;
+
+      btnCb(installed);
+
+      if (installed) {
+        this.reload();
+      }
+    },
+
+    async updateHarvesterExtension(btnCb) {
+      let updated = false;
+
+      try {
+        let harvesterRepository = this.harvesterRepository;
+
+        if (this.harvester.missingRepository) {
+          harvesterRepository = await ensureHelmRepository(this.$store, HARVESTER_REPO.spec.gitRepo, HARVESTER_REPO.metadata.name, HARVESTER_REPO.spec.gitBranch);
+
+          await this.setHarvesterLatestVersion(harvesterRepository);
+        }
+
+        await installHelmChart(harvesterRepository, { ...HARVESTER_CHART, version: this.harvesterLatestVersion }, {}, UI_PLUGIN_NAMESPACE, 'upgrade');
+
+        const extension = await waitForUIExtension(this.$store, HARVESTER_CHART.name);
+
+        updated = await waitForUIPackage(this.$store, { ...extension, version: this.harvesterLatestVersion });
+      } catch (error) {
```

Comment:
Do we need to set any error in catch block ? 
Same for installHarvesterExtension()